Merge previous changes

RuhaniMittal29 opened the pull request:
## Description

Merge the new-dev branch with the production branch

## Updates



## Checklist

- [x] Ran `npm run lint:fix` to format code
- [x] Requested at least one reviewer
- [x] Connected PR to website team board ticket
- [x] Addressed code review comments
- [ ] Gained at least one approval for your PR

Commit:
Finished navbar

ryanyae committed:
fixed

MuzammilC committed:
Initialized Development

MuzammilC committed:
Made Laptop View of Club Intro

MuzammilC committed:
Added divider

MuzammilC committed:
Added config files

MuzammilC committed:
Removed files from commit

MuzammilC committed:
Added the card icon

ryanyae committed:
Fixed routes and static files

Commit:
feat(src/pages/HomeV2/Home.ts): Add scss and initial setup of hero

alim0118 committed:
Added team into text to static file

alim0118 committed:
Set template (title and body without styling)

diegoAyoub committed:
Responsive design. All sizes. Small bug @~1100px

ryanyae committed:
Merge pull request #17 from ubcdsci/connor-1-navbar

Finished navbar

diegoAyoub committed:
Finished footer

Commit:
feat: add styles to content

MuzammilC committed:
Merge branch 'new-development' into muzammil-4-club-intro

MuzammilC committed:
Merge pull request #18 from ubcdsci/muzammil-4-club-intro

Muzammil 4 club intro

alim0118 committed:
FInished desktop version

ryanyae committed:
Fixed

Commit:
feat: finish hero section + css

MuzammilC committed:
Merge branch 'new-development' into diego-6-footer

MuzammilC committed:
Merge pull request #19 from ubcdsci/diego-6-footer

Diego 6 footer

MuzammilC committed:
Fixed text gradient

MuzammilC committed:
Merge branch 'new-development' into adrian-2-hero

MuzammilC committed:
Merge pull request #21 from ubcdsci/adrian-2-hero

Adrian 2 hero

MuzammilC committed:
Fixed some minor issues

MuzammilC committed:
revert logo sizing

MuzammilC committed:
Made changes to increase responsiveness

MuzammilC committed:
updated placeholder image

MuzammilC committed:
Added the club into section to home

MuzammilC committed:
fixed text padding

MuzammilC committed:
resposive changes

MuzammilC committed:
made more responsiveness changes

MuzammilC committed:
Merge branch 'new-development' into origin/annabel-3-teamintro

MuzammilC committed:
Merge pull request #20 from ubcdsci/origin/annabel-3-teamintro

Origin/annabel 3 teamintro

MuzammilC committed:
Added the Team intro section to home

MuzammilC committed:
Added material ui package

MuzammilC committed:
Made the club signup section

MuzammilC committed:
made ui changes

MuzammilC committed:
Merge pull request #22 from ubcdsci/muzammil-club-sign-up

Muzammil club sign up

MuzammilC committed:
Made the upcoming events section

MuzammilC committed:
small changes

MuzammilC committed:
Merge pull request #23 from ubcdsci/muzammil-upcoming-event

Made the upcoming events section

MuzammilC committed:
Made the past event cards

MuzammilC committed:
Merge pull request #24 from ubcdsci/muzammil-past-events

Made the past event cards

MuzammilC committed:
made more changes to the past project

loganihrishi committed:
fix css & styling for PastEvents

siddiskid committed:
fixed intro section to include three cards

ubcdsciclub committed:
Merge pull request #35 from ubcdsci/muzammil-past-projects

Added project page to the website

RuhaniMittal29 committed:
changed the colour scheme

RuhaniMittal29 committed:
minor changes

MuzammilC committed:
Merge pull request #33 from ubcdsci/past_events_component_Hrishi_Logani

fix css & styling for PastEvents

MuzammilC committed:
Merge pull request #34 from ubcdsci/siddarth-7-fix-intro

fixed intro section to include three cards

MuzammilC committed:
Merge pull request #36 from ubcdsci/projects

changed the colour scheme

Brian-Soh committed:
Contact Us Page

isobela committed:
remove commented out code

MuzammilC committed:
Fixed deployment issues

MuzammilC committed:
fixed build issues

MuzammilC committed:
Merge pull request #43 from ubcdsci/muzammil-build-issues-fix

fixed build issues

isobela committed:
remove old home component

MuzammilC committed:
Merge pull request #42 from ubcdsci/isobel-9-cleanhome

Clean up home page

Brian-Soh committed:
Updated favicon and error page logo

RuhaniMittal29 committed:
Merge branch 'production' into new-development

MuzammilC commented on src/components/ClubInto/index.tsx:
```diff
@@ -0,0 +1,88 @@
+            <div className={styles.ProjectIcon} style={{'color': `${data[props.index].color}`}} >
+                {data[props.index].icon}
+            </div>
+            <ProjectButton text={data[props.index].button} href="/" color={data[props.index].color}/>
```
This button does not link the cards to the respective pages. Figure out a way to make the href dynamically point to the respective cards pages.

MuzammilC commented on src/components/SignUp/SignUp.tsx:
```diff
@@ -0,0 +1,32 @@
+        <span className={styles.SignUpButton}>
+            <GreenButton
+                    text="Join Us"
+                    href="https://ubc.ca1.qualtrics.com/jfe/form/SV_6VTh44IkLtOyjIy"
```
Update this link so that it points to our current membership form. Make this dynamic instead of static so that it is easier to change in the future.

MuzammilC commented on src/components/NavBar/index.tsx:
```diff
@@ -176,27 +169,31 @@ const NavBar = () => {
-                text="Register Now"
-                href="https://ubc.ca1.qualtrics.com/jfe/form/SV_8j2JFMxAQFnBlki"
+                text="REGISTER"
+                href="https://ubc.ca1.qualtrics.com/jfe/form/SV_6VTh44IkLtOyjIy"
```
Update this link so that it is the current registration form. Also make this dynamic so that it is easier to update this link in the future.

MuzammilC commented on src/components/HeroSection/index.tsx:
```diff
@@ -0,0 +1,25 @@
+      </p>
+      <div className={styles.EmailRegister}>
+        <input type="text" placeholder="EMAIL" className={styles.EmailInput} />
+        <button className={styles.RegisterButton}>Register</button>
```
What is the use of this register button?

MuzammilC commented on src/components/Footer/index.tsx:
Update all the content in the footer and make sure that the links direct to the proper parts of the page and not just the top of the page.

MuzammilC commented on src/pages/ProjectsV2/ProjectsV2.tsx:
Update all of the content on this pages so that it has all the current content.

MuzammilC commented on src/pages/EventsV2/Event.tsx:
Update all the content on the events Page.

Commit:
Address PR #55 review: config-driven links, hero fix, and responsive styling

- Add shared MEMBERSHIP_FORM_URL constant in config and use it across
  SignUp, NavBar (mobile + desktop), HeroSection, and FAQ data
- Add href field to ClubIntroCardData so card buttons link to their
  respective pages dynamically (/projects, /events, /contact-us)
- Replace non-functional email input and register button in HeroSection
  with a working link to the membership form
- Fix Footer column title links to use smooth scrolling instead of
  forcing scrollTop, so hash-based navigation works correctly
- Adjust HeroSection and TeamIntro padding/font-size for smaller screens
  to prevent overlap and improve readability
- Flag placeholder content in Events and Projects data for team update

Made-with: Cursor

ubcdsciclub committed:
Merge pull request #63 from ubcdsci/fix/pr-55-review

Address PR #55 review: config-driven links, hero fix, and responsive styling
